src/views/bouquets/BouquetEditView.vue

```diff
@@ -0,0 +1,147 @@
+<script setup>
+import { ref, onMounted } from "vue"
+import { useDatasetStore } from "../../store/DatasetStore"
+import { useBouquetStore } from "../../store/BouquetStore"
+import { useRouter, useRoute } from "vue-router"
+import SearchAPI from "../../services/api/SearchAPI"
+import Multiselect from "@vueform/multiselect"
+
+const searchAPI = new SearchAPI()
+const datasetStore = useDatasetStore()
+const bouquetStore = useBouquetStore()
+const router = useRouter()
+const route = useRoute()
+
+const isCreate = route.name === "bouquet_add"
+
+const form = ref({})
+const datasets = ref([])
+const selectedDataset = ref(null)
+const selector = ref(null)
+const loadedBouquet = ref({})
+
+const search = async (query) => {
+  if (!query) return []
+  const results = await searchAPI._search(query, { hitsPerPage: 10 })
+  return results.hits.map(r => {
+    return { value: r.id, label: r.title }
+  }).filter(r => !datasets.value.map(d => d.id).includes(r.value))
+}
+
+const onSelect = async (value) => {
+  const dataset = await datasetStore.load(value)
```

Comment:
I get an error when selecting a dataset that doesn't exist on data.gouv (I'm guessing search index lag).
For instance, search "eau" and select "Donnée Sécheresse - Propluvia" => https://demo.data.gouv.fr/api/1/datasets/6470b39cfbad66b8c265ada3/ isn't found and breaks.

Reply:
Right, let's say error handling has not been a priority on this POC 😅 I fixed it to not crash the whole app when it happens. BTW I can't reproduce this way, I've reindexed on demo and I don't have this dataset.

Reply:
Yeah, not crashing the whole app sounds like a good middle ground ;)
I suspected my exemple would be short lived, good thing it was easy to diagnose